### ISSUE TITLE
Proton 4.11-7 Elite Dangerous - Recognizing Logitech Extreme 3D Pro Joystick as 'Generic Gamepad'

### ISSUE THREAD

**Issue by LauritsLL** [ValveSoftware/Proton#3188]
I switched from W10 to Ubuntu 18.04 LTS some days ago, and up until now, I have found solutions for all my noob questions about stuff.

But recently, I installed Steam and tried to install Elite Dangerous: Horizons. I got it up and working on Proton 4.11-7 with the mouse, but I had always been using my trusty Logitech Extreme 3D Pro Joystick on Windows 10 when playing the game, but when trying to set up the joystick on Ubuntu, I immediately ran into a lot of problems. When I look at 'General Controller Settings' I can see that Steam for some reason thinks that my joystick is a Generic Gamepad ??

_Picture of General Controller Settings in Steam_
https://i.stack.imgur.com/LS6Yb.png

I thought that that was pretty weird, and Elite Dangerous thinks the same too when I fire up the game. Steam thinks that my joystick is a Generic Gamepad and I don't know why?

**What I've tried:**
1. I've tried re-installing Elite Dangerous. No effect.
2. I've tried setting the 'Steam Input Per-Game Setting' to 'Forced Off'. No effect.
3. Tested the joystick with jstest-gtk. It is recognizing it correctly as a joystick!
4. I also tried launching ED in Big Picture Mode and tried Steam Beta Release... No effect at all.

Please help. Any help will be highly appreciated!

**Comment by Naia-love** [comment]
Hey, you need to set up some drivers before use it. I searched a bit, and all I found is some command.
So try to run those command:
```
modprobe joydev
dmesg | grep usb
```
I also found [this](https://www.infradead.org/~mchehab/kernel_docs/input/joystick.html)

**Comment by LauritsLL** [comment]
> Hey, you need to set up some drivers before use it. I searched a bit, and all I found is some command.
> So try to run those command:
> 
> ```
> modprobe joydev
> dmesg | grep usb
> ```
> 
> I also found [this](https://www.infradead.org/~mchehab/kernel_docs/input/joystick.html)

Did everything you said and tried every loading every module in the web page...
Still same result. :/
`sudo modprobe joydev ns558 serport analog`

Also the `dmesg | grep usb` says this.
`[   97.250817] usb 1-2: New USB device strings: Mfr=1, Product=2, SerialNumber=0
[   97.250819] usb 1-2: Product: Logitech Extreme 3D
[   97.250821] usb 1-2: Manufacturer: Logitech
[   97.272724] input: Logitech Logitech Extreme 3D as /devices/pci0000:00/0000:00:14.0/usb1/1-2/1-2:1.0/0003:046D:C215.0007/input/input30
[   97.331868] logitech 0003:046D:C215.0007: input,hidraw3: USB HID v1.10 Joystick [Logitech Logitech Extreme 3D] on usb-0000:00:14.0-2/input0
[  122.267917] usb usb4: USB disconnect, device number 1
[  122.268224] usb usb3: USB disconnect, device number 1`

But for some _weird reason, Steam_ thinks that my "Joystick" is a "Generic Gamepad"....

**Comment by LauritsLL** [comment]
Also when I do `jstest /dev/input/js0` it is recognizing it correctly and interpreting the values correctly.

**Comment by redmcg** [comment]
@Lause132 I don't know if you ever resolved this issue - but I know people have had success with the following:
https://github.com/redmcg/wine/wiki/Peripherals#logitech-extreme-3d-pro

**Comment by GummyJon1** [comment]
@redmcg I think it's better to list the instructions you said over at the following, since it goes more in detail for newbies (such as myself) https://askubuntu.com/questions/1185679/steamplay-elite-dangerous-recognizing-logitech-extreme-3d-pro-joystick-as-gen?rq=1. It is basically the same, but just more detailed. I can confirm this fix works, however.

**Comment by redmcg** [comment]
@GummyJon1 Thanks for the suggestion. I've updated the Wiki

**Comment by LauritsLL** [comment]
Hello everyone - I changed back to Windows shortly after I created this issue, because of school and because it was getting so frustrating.
I'm really sorry that you all had to wait for this issue to be closed, but as I can see @GummyJon1 found my Stackoverflow question of this issue and has confirmed that the answer has fixed the issue. :)
As there now is a fix/workaround I will close this issue.
Thank you for everyone's help.
